docs/use_cases/recommender_systems.md

```diff
@@ -1,15 +1,111 @@
 
 # Recommender Systems
 
-<!-- TODO: Cover image: 
-1. You can create your own cover image and put it in the correct asset directory,
-2. or you can give an explanation on how it should be and we will help you create one. Please tag arunesh@superlinked.com or @AruneshSingh (GitHub) in this case. -->
+![](assets/use_cases/recommender_systems/recommender.jpg)
 
-your content here
+Recommender Systems are becoming increasingly important, given the plethora of products offered to the users/customers. For example, in early days, fashion retailers developed basic versions of content-based recommenders that served quite well, but soon the need emerged for more relevant recommendations that could increase user’s engagement. At the moment of this writing, fashion retailers use far more sophisticated approaches that make use not only of the user’s purchasing/viewing history, but also of user and item metadata: user age, location, spending habits, mood, etc.; item category, popularity, etc.
+
+However, it is sometimes the case that only scarce side information about the users or the items is available to us. Consider, for example, a public service provider that has its own audio- and video-on-demand platform; such companies have restrictions with regard to collecting/storing user metadata. But even in this case, we still have useful information about the items that are served, such as genre, popularity, etc. Still, developers often disregard this side information, because it is scarce, and proceed with collaborative-filtering (CF) approaches, which make use of the historical interactions only (data consisting of user-item pairs). While CF work reasonably well in this use case - extrapolate user preferences via similarities of all-user’s browsing/purchasing history - one question remains: can we improve on the quality by (somewhow) making use of available information, regardless how scarce? More precisely, can we use the collaborative filtering paradigm in this case, with this scarce side info? Yes, there are libraries that allow us to ‘inject’ side information ([LightFM](https://making.lyst.com/lightfm/docs/home.html), for example). But there is also a way to extend the most efficient and effective collaborative filtering models, such as [ALS Matrix Factorization (MF)](http://yifanhu.net/PUB/cf.pdf) (the Python [‘implicit’](https://github.com/benfred/implicit) library) or EASE, to make use of the side information.
```

Comment:
I am not sure I understand this part:

I also can't tell if ALS Matrix Factorization or EASE [you should spell out what EASE stands for, since you're doing the same with MF] is a way of extending collaborative filtering, or it's simply a better collaborative filtering model?
If your main point is that you can use collaborative filtering mechanisms, but in order to maximize recommendation quality, you also have to EXTEND collaborative filtering mechanisms, then you should provide examples of extending collaborative filtering models.
Are MF and EASE part of a general category? Or are they the only two ways of extending collab filtering models?
If general category, introduce them that way, as part of a general category.
----
ALS and EASE are collaborative filtering models, and I am showing how their capabilities an be improved by integrating additional data. Some suggestion on how and where I can make this clearer?

Reply:
Instead of saying ~ "there are collaborative filtering models.... but..."
say "there are collaborative filtering models.... and, furthermore, ..."
This makes clear that they are part of the same solution... and not only do we have solution; it is a very good solution
rather than, we can use collab filtering model... but it's better if we extend these models (i.e., the latter is the real solution, whereas the former is not really much of a solution). "but" generally implies an opposition.